Frameworks/CoreFoundation/URL.subproj/CFURL.c

```diff
@@ -5007,21 +5007,35 @@ CFURLRef CFURLCreateCopyDeletingPathExtension(CFAllocatorRef allocator, CFURLRef
     }
 #endif
     
+    CFStringRef lastPathComponent = NULL;
     rg = _rangeOfLastPathComponent(url);
-    if (rg.location < 0) {
+    if ( (rg.location == kCFNotFound) || ((lastPathComponent = CFStringCreateWithSubstring(allocator, url->_string, rg)) == NULL) ) {
+        // no last path component
         result = NULL;
-    } else if (rg.length && CFStringFindWithOptions(url->_string, CFSTR("."), rg, kCFCompareBackwards, &dotRg)) {
-        CFMutableStringRef newString = CFStringCreateMutableCopy(allocator, 0, url->_string);
-        dotRg.length = rg.location + rg.length - dotRg.location;
-        CFStringDelete(newString, dotRg);
-        result = _CFURLCreateWithArbitraryString(allocator, newString, url->_base);
-        CFRelease(newString);
-    } else {
-        result = (CFURLRef)CFRetain(url);
     }
```

Comment:
clang-format is turned off for CF to minimize churn but we should be diligent ourselves anyway.

---
In reply to: [99636921](https://github.com/Microsoft/WinObjC/pull/1906#discussion_r99636921) [](ancestors = 99636921,99223362)